### PR TITLE
Create separate Custom Roles for E2E

### PR DESCRIPTION
**Pull request by karolkieglerski**

### CODE REVIEW

File: tests/gcp_iam_roles/dynatrace-gcp-function-e2e-deployment-role.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  -  container.pods.list
+  -  container.secrets.create
+  -  container.secrets.delete
+  -  container.secrets.get
+  -  container.secrets.list
+  -  container.secrets.update
+  -  container.serviceAccounts.create
+  -  container.serviceAccounts.delete
+  -  container.serviceAccounts.get
+  -  logging.sinks.create
```
Comment (sq2gxo) [inline tests/gcp_iam_roles/dynatrace-gcp-function-e2e-deployment-role.yaml:23]: sink permissions are not needed for helm deployment itself, should be moved to e2e-test role

File: tests/gcp_iam_roles/dynatrace-gcp-function-e2e-tests-role.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - cloudfunctions.functions.invoke
+  - cloudfunctions.functions.list
+  - cloudfunctions.functions.sourceCodeGet
+  - cloudfunctions.functions.sourceCodeSet
+  - cloudfunctions.functions.update
+  - cloudfunctions.locations.get
+  - cloudfunctions.locations.list
+  - cloudfunctions.operations.get
+  - cloudfunctions.operations.list
+  - iam.roles.create
```
Comment (sq2gxo) [inline tests/gcp_iam_roles/dynatrace-gcp-function-e2e-tests-role.yaml:18]: iam permissions are needed for helm deployment - add them also to deployment role

File: gcp_iam_roles/dynatrace-gcp-function-deployment-role.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# create with command:
+# gcloud iam roles create dynatrace_function.deployment --project={PROJECT_ID} --file=dynatrace-gcp-function-deployment-role.yaml
+title: Dynatrace GCP Function deployment role
+description: Role for Dynatrace GCP function helm and pubsub deployment
+stage: GA
+includedPermissions:
+  - pubsub.subscriptions.get
```
Comment (sq2gxo) [inline gcp_iam_roles/dynatrace-gcp-function-deployment-role.yaml:7]: from pubsub set of permissions seem like only   pubsub.subscriptions.get (maybe also pubsub.subscriptions.list) is needed for deploy-helm script.
Other pubsub permissions should be moved into e2e-tests role

Comment (karolkieglerski) [reply gcp_iam_roles/dynatrace-gcp-function-deployment-role.yaml:7]: no, all of this are required. Our documentation has two files `deploy-helm.sh` and `deploy-pubsub.sh`

Comment (karolkieglerski) [reply gcp_iam_roles/dynatrace-gcp-function-deployment-role.yaml:7]: that other settings are required by that second script. It was tested

**Review by sq2gxo — DISMISSED** [review]

**Review by sq2gxo — APPROVED** [review]

**Review by Dawid-Kaszubski — APPROVED** [review]

**Review by Dawid-Kaszubski — APPROVED** [review]

**Review by mswiatkowska — APPROVED** [review]

**Pull request merged**